lldb/source/Plugins/ObjectFile/Mach-O/ObjectFileMachO.cpp

```diff
@@ -6519,15 +6519,13 @@ struct page_object {
 };
 
 bool ObjectFileMachO::SaveCore(const lldb::ProcessSP &process_sp,
-                               const FileSpec &outfile,
-                               lldb::SaveCoreStyle &core_style, Status &error) {
-  if (!process_sp)
+                               const lldb_private::CoreDumpOptions &core_options,
+                               Status &error) {
+  auto core_style = core_options.GetCoreDumpStyle();
+  const auto outfile = core_options.GetOutputFile();
+  if (!process_sp || !outfile)
     return false;
 
-  // Default on macOS is to create a dirty-memory-only corefile.
-  if (core_style == SaveCoreStyle::eSaveCoreUnspecified)
-    core_style = SaveCoreStyle::eSaveCoreDirtyOnly;
-
```

Comment:
This is a functionality change for ObjectFileMachO.cpp since we now default to stack only in `lldb_private::CoreDumpOptions`. @jasonmolenda you ok if we switch the default for mach-o to stack only? I would like to move the default decision into common code so different core file formats don't each pick a different default if possible.